### DISCUSSION TITLE
Pass click of a dismissable popup

### DISCUSSION THREAD

**Discussion by c33howard** [unpoly/unpoly#291] [Q&A]
I've created some dismissable popups using unpoly.  It's working great!  In some cases, I want a click outside the popup to close the popup, but also pass the click to the event target (note: this isn't just on anchors, but I've got some logic around click on canvas elements).  Is there a nice way to do this within unpoly?  I'm currently registering a click handler, doing event.stopPropagation() and layer.dimiss() by hand.  Feels like there should be a cleaner way.

Thanks,
Craig

**Comment by adam12** [comment]
Could you make an example of what you have so far and are trying to cleanup? It would be easier for us to follow along. You can remix this [Glitch project](https://glitch.com/edit/#!/empty-unpoly-project) as a starter.

**Reply by c33howard** [reply]
Cool.  I hadn't seen glitch before.

https://glitch.com/edit/#!/prism-horse-ferry

Instructions for repro are in index.html.  I shrunk the example, but it still feels a bit big.  I had to use javascript to trigger the behaviour.

**Comment by triskweline** [comment]
Unpoly will stop and prevent any event that it handles. There is no way to configure this differently.

You can probably make this work by listening to `up:layer:dismissed`, but Unpoly popups are not a good tool for implementing tooltips that appear on hover. 

Unpoly popups were built for drop-down menus that open on click. They can host arbitrary content and multi-page interactions. A good use case would be selecting a billing address from a dropdown, but also allow creating a new address without leaving the page.

There are many tiny solutions that do this better, including [some that work with pure CSS](https://microtip.vercel.app/).

**Reply by c33howard** [reply]
Why do you say that unpoly is not a good tool for tooltips?  Other than this click issue, it's working great.

The example was overly simplistic.  Some of my tooltips include images.  Some have a button.  This maybe a terrible design choice in general, but it's working well for what I'm trying to accomplish.
